src/go/k8s/api/cluster.redpanda.com/v1alpha1/topic_types.go

```diff
@@ -119,6 +135,178 @@ type KafkaAPISpec struct {
 	SASL *KafkaSASL `json:"sasl,omitempty"`
 }
 
+func (k *KafkaAPISpec) ConfigureTLS(ctx context.Context, namespace string, cl client.Client, opts []kgo.Opt, log logr.Logger) ([]kgo.Opt, error) {
```

Comment:
This and `ConfigureSASL` were just extracted from the `internal/controller/cluster.redpanda.com` subpackage to publicly expose them and make them reusable.